.github/workflows/phpunit-tests.yml

```diff
@@ -0,0 +1,144 @@
+name: PHPUnit Tests
+
+# Since Unit Tests are required to pass for each PR,
+# we cannot disable them for documentation-only changes.
+on:
+    pull_request:
+    push:
+        branches:
+            - trunk
+            - 'release/**'
+            - 'wp/**'
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+    # The concurrency group contains the workflow name and the branch name for pull requests
+    # or the commit hash for any other events.
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    # Runs the PHPUnit tests for Gutenberg.
+    #
+    # Performs the following steps:
+    # - Sets environment variables.
+    # - Checks out the repository.
+    # - Sets up Node.js.
+    # - Sets up PHP.
+    # - Installs Composer dependencies.
+    # - Installs npm dependencies
+    # - Logs general debug information about the runner.
+    # - Logs Docker debug information (about the Docker installation within the runner).
+    # - Starts the wp-env Docker containers.
+    # - Logs the running Docker containers.
+    # - Logs debug information about what's installed within the wp-env Docker containers.
+    # - Install WordPress and Gutenberg within the Docker container.
+    # - Run the PHPUnit tests.
+    test-php:
+        name: ${{ matrix.php }}${{ matrix.multisite && ' multisite' || '' }} on ${{ matrix.os }}
+        runs-on: ${{ matrix.os }}
+        timeout-minutes: 20
+        if: ${{ github.repository == 'WordPress/gutenberg' || github.event_name == 'pull_request' }}
+        strategy:
+            fail-fast: false
```

Comment:
Is there a reason to not fail fast ? Removing this saves a lot of resources.

Reply:
My reasoning for setting it to `false` is the following:
It's convenient to see all CI jobs on which tests are failing.
Often unit tests fail on some PHP versions while passing on others.
If the `fail-fast` option is true, GitHub will show only the first failed job.

Reply:
I totally get the convenience argument.
Having said that, in my experience, more often than not, when there is a tests failure, it fails on all PHP versions.
Of course, there are times when it is PHP version specific and in those cases, the runtime of the test runs will end up deciding when the build fails and what shows, which is not always helpful.

I'm not sure what type of GH account the WP organisation has, but I find that in most cases these (counter-)arguments apply:
* If it's a free account: be kind to GH, which is shouldering the costs, by only using the resources we need.
* If it's a paid account: be kind to accounting as every build counts towards the minutes billed.
* In both cases: be kind to the environment and don't use more server resources than are strictly needed.

Let's hear from some more people/opinions ?

Reply:
I've found an interesting [Trac](https://core.trac.wordpress.org/ticket/52612) ticket.
It explains why the `fail-fast` option was disabled in Core.
I thought Gutenberg should profess the same principles.

> If it's a free account: be kind to GH, which is shouldering the costs, by only using the resources we need.

As far as I understand this repo is far from hitting the usage limits, as each job is allowed [to run for 6 hours](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits).

> Let's hear from some more people/opinions ?

Yes, it would be interesting to hear more opinions on this matter.
Please feel free to tag anyone who could share an opinion on this matter.
Thanks!
Speaking of me, I still see value in having it disabled because of the reasons above.

Reply:
I've found an interesting [Trac](https://core.trac.wordpress.org/ticket/52612) ticket.
It explains why the `fail-fast` option was disabled in Core.
I thought Gutenberg should profess the same principles.

> Let's hear from some more people/opinions ?
Yes, we can do it.
But if it takes long to get someone else to review this, I'd prefer to enable fast failing now and disable it later if it causes issues (as it did in Core).

Reply:
I've enabled fast-fail as there were no more comments.
9452534594cc5e60b9b35b83e197fdd6a251c317